src/decoder/decodeqrdata.ts

```diff
@@ -219,7 +219,7 @@ function decodeByteSegment(bits: BitStream, result: resultByteArray, count: numb
   for (var i = 0; i < count; i++) {
     readBytes[i] = bits.readBits(8);
   }
-  result.val = readBytes
+  Array.prototype.push.apply(result.val, readBytes);
```

Comment:
Oh yikes - really. good catch. I forgot this code was mutating result instead of being more pure.